static/js/util/profile_edit_test.js

```diff
@@ -391,35 +360,52 @@ describe('Profile Editing utility functions', () => {
     });
   });
 
-  describe("Bound month/year field", () => {
-    let monthYearField, monthTextField, yearTextField;
-    let validateYearSpy, validateMonthSpy;
-    let renderMonthYearField = () => {
-      monthYearField = boundMonthYearField.call(
+  describe("Bound date field", () => {
+    let dateField, dateTextField, monthTextField, yearTextField;
+    let validateYearSpy, validateMonthSpy, validateDateSpy;
+    let renderDateField = (...args) => {
+      dateField = boundDateField.call(
         that,
         ["date_of_birth"],
-        "Date of birth"
+        "Date of birth",
+        ...args
       );
 
-      return monthYearField.props.children.filter(React.isValidElement);
+      return dateField.props.children.filter(React.isValidElement);
     };
 
     beforeEach(() => {
-      [monthTextField, , yearTextField] = renderMonthYearField();
-
       validateYearSpy = sandbox.spy(util, 'validateYear');
       validateMonthSpy = sandbox.spy(util, 'validateMonth');
+      validateDateSpy = sandbox.spy(util, 'validateDate');
+    });
+
+    afterEach(() => {
+      yearTextField = null;
+      monthTextField = null;
+      dateTextField = null;
     });
 
+    let rerender = omitDate => {
+      if (omitDate) {
+        [monthTextField, , yearTextField] = renderDateField(true);
+      } else {
+        [monthTextField, , dateTextField, , yearTextField] = renderDateField(false);
+      }
+    };
+
     it("has proper props for an invalid or missing value", () => {
       for (let dateOfBirth of ['', {}, null, undefined]) {
         that.props.profile.date_of_birth = dateOfBirth;
-        [monthTextField, , yearTextField] = renderMonthYearField();
+        rerender(false);
 
         assert.equal(monthTextField.props.floatingLabelText, "Date of birth");
```

Comment:
If you look above you can see it being used on the education and employment tabs of the profile (in `EmploymentTab.js` and `EducationTab.js`)